cpp/daal/src/algorithms/svm/oneapi/svm_helper_oneapi.h

```diff
@@ -168,6 +171,9 @@ struct HelperSVM
         services::internal::sycl::KernelArguments args(4, status);
         DAAL_CHECK_STATUS_VAR(status);
 
+        DAAL_ASSERT(indX.size() == nWS);
+        DAAL_ASSERT(newX.size() == nWS * p);
```

Comment:
is overflow check needed here?

Reply:
It should checked when buffer is allocate. For my case it [here](https://github.com/oneapi-src/oneDAL/blob/master/cpp/daal/src/algorithms/svm/oneapi/svm_train_cache_oneapi.h#L194).

Reply:
It is not seen that it is safe.By the reference the numeric table is allocated but not a buffer. How you guarantee the safety of that function if some changes in the code that calls it? E.g. it is called with different block. Place a comment here at least why you think overflow cannot happen here